internal/service/clusteroutagesimulation/resource_cluster_outage_simulation.go

```diff
@@ -214,38 +213,38 @@ func resourceClusterOutageSimulationRefreshFunc(ctx context.Context, clusterName
 	}
 }
 
-func convertOutageSimulationToSchema(outageSimulation *matlas.ClusterOutageSimulation, d *schema.ResourceData) error {
-	if err := d.Set("state", outageSimulation.State); err != nil {
+func convertOutageSimulationToSchema(outageSimulation *admin.ClusterOutageSimulation, d *schema.ResourceData) error {
+	if err := d.Set("state", outageSimulation.GetState()); err != nil {
 		return fmt.Errorf(errorClusterOutageSimulationSetting, "state", err)
 	}
-	if err := d.Set("start_request_date", outageSimulation.StartRequestDate); err != nil {
+	if err := d.Set("start_request_date", conversion.TimeToString(outageSimulation.GetStartRequestDate())); err != nil {
```

Comment:
```suggestion
	if err := d.Set("start_request_date", conversion.TimePtrToStringPtr(outageSimulation.StartRequestDate)); err != nil {
```

I you look at TimeToString implementation it will fail (or render a weird time) if it's nil, better to work with the pointer function

Reply:
I think it will panic because UTC doesn't check nil:

// UTC returns t with the location set to UTC.
func (t Time) UTC() Time {
	t.setLoc(&utcLoc)
	return t
}

Reply:
Thanks for noticing, changed to use `TimePtrToStringPtr`